### ISSUE TITLE
As I site administrator I want to see a list of registered users

### ISSUE THREAD

**Issue by lcabrini** [AluguntuguiProject/aluguntugui#9]

**Comment by benbaka** [comment]
This story assumes that we have the role "Site Administrator" already. Let's create other stories that cover and precede it:
1. As a visitor, I should be able to select which user role I belong to during registration.
2. As a visitor, I should see the roles "Site Administrator" and "Regular User" as options during registration.

You may augment and add the relevant stories.

**Comment by lcabrini** [comment]
User stories DO assume roles that would have to exist or be created accordingly. If the Site Administrator role does not exist, one of the dependent tasks would be to create this role. Remember, user stories are about capturing functionality.

I possible precedent story would be "as a site administrator I want to create a new role". Another useful story may be "as a site administrator I want to assign a role to a user". 

However, the site administrator role as such would be a primary role. That is this role must exist when the system is first started. Also, there must be a user which has the role site administrator when the system first starts, as that is the very user responsible for setting up the system.

It would probably not be wise to allow users to select their roles at registration time. I can image that the vast majority of users would then choose to be site administrators.

**Comment by lcabrini** [comment]
Anyways, you do have a very good point there. We should really have a place where we keep a track of the roles that are pre-defined in the system. From the stories I have written so far, I can see three.

1. visitor: somebody who has not yet registered on the site
2. user: somebody who has registered and is able to learn languages
3. site administrator: somebody who manages the site, blocks/unblocks users, approves etc.

I think there may be other roles eventually, but for now, these are the ones that come to mind.

**Comment by lcabrini** [comment]
I propose this story to be added to the sprint backlog for sprint 1.

**Comment by benbaka** [comment]
Agreed.

**Comment by benbaka** [comment]
👍 Move to Sprint Backlog

**Comment by benbaka** [comment]
A few things come to mind with respect to the Site Administrator.

1. If any other user type accesses his page, they must be returned to an unauthorized page with notice. 
2. In the site administrators view, he can also see all users. 
3. There must be an existing group in the system (using seeds at the time of installation) to which the first site administrator is tied to. 
4. In his view he can search for a user and assign a different role. 

Let's flesh out more preceding user stories into the "Proposed" board.

**Comment by lcabrini** [comment]
I agree. I also think that the admins view should contain user admin functions in addition to being a list, such as block/unblock.
